src/webots/widgets/WbMessageBox.cpp

```diff
@@ -69,6 +69,14 @@ void WbMessageBox::critical(const QString &message, QWidget *parent, const QStri
     WbLog::error(message);
 }
 
+void WbMessageBox::fatal(const QString &message, QWidget *parent, const QString &title) {
+  if (gEnabled)
+    QMessageBox::critical(parent, title, message);
+  else
+    WbLog::error(message);
```

Comment:
Instead of duplicating the code you can call `WbMessageBox::critical`.

Reply:
My concern is that WbMessageBox::critical does not quit Webots.
I just remark it's the purpose of WbLog::fatal.
If it's ok I will drop my modifications and use WbLog::fatal instead.